### ISSUE TITLE
NuGet.exe restore <sln> pops assertion dialog

### ISSUE THREAD

**Issue by AArnott** [NuGet/Home#1633]
nuget Version: 3.2.0.10516 (latest public version)

```
git clone https://github.com/AArnott/pinvoke.git
cd pinvoke\src
git checkout 0060f4de9826
nuget restore
```

Among some text output, this assertion failed dialog appears:

```
---------------------------
Assertion Failed: Abort=Quit, Retry=Debug, Ignore=Continue
---------------------------

   at NuGet.Frameworks.FrameworkNameProvider.TryGetPortableFrameworks(String shortPortableProfiles, IEnumerable`1& frameworks)
   at NuGet.Frameworks.FrameworkNameProvider.TryGetPortableFrameworks(String profile, Boolean includeOptional, IEnumerable`1& frameworks)
   at NuGet.Frameworks.CompatibilityProvider.IsPCLCompatible(NuGetFramework target, NuGetFramework candidate)
   at NuGet.Frameworks.CompatibilityProvider.IsCompatibleCore(NuGetFramework target, NuGetFramework candidate)
   at NuGet.Frameworks.CompatibilityProvider.<>c__DisplayClass5_0.<IsCompatible>b__0(Int32 key)
   at System.Collections.Concurrent.ConcurrentDictionary`2.GetOrAdd(TKey key, Func`2 valueFactory)
   at NuGet.Frameworks.CompatibilityProvider.IsCompatible(NuGetFramework target, NuGetFramework candidate)
   at NuGet.Frameworks.FrameworkReducer.<>c__DisplayClass7_0.<GetNearestInternal>b__3(NuGetFramework f)
   at System.Linq.Enumerable.<>c__DisplayClass0`1.<CombinePredicates>b__1(TSource x)
   at System.Linq.Enumerable.Wh......
<truncated>
---------------------------
Abort   Retry   Ignore   
---------------------------
```

**Comment by AArnott** [comment]
It appears to be due to an unexpected value in my project.json file:

``` json
{
  "supports": { },
  "dependencies": {
    "xunit": "2.1.0",
    "xunit.runner.visualstudio": "2.1.0"
  },
  "frameworks": {
    ".NETPortable,Version=v4.6": { }
  },
  "runtimes": {
    "win-anycpu": { }
  }
}
```

Here I have that I'm targeting `.NETPortable,Version=v4.6` which is incorrect. Changing it to `.NETFramework,Version=v4.6` resolved the issue. Nevertheless an error message at the command line seems better than a blocking assertion dialog in retail software.

**Comment by emgarten** [comment]
This has been fixed in 3.2.1-rc
